src/js/components/RadioButtonGroup/RadioButtonGroup.js

```diff
@@ -101,7 +101,20 @@ const RadioButtonGroup = forwardRef(
       }, 1);
     };
 
+    const onRadioButtonChange = (event, optionValue) => {
+      setValue(optionValue);
+      if (onChange) {
+        event.persist(); // extract from React synthetic event pool
+        
+        // event.target.value gives value as a string which needs to be
+        // manually typecasted according to the type of original option value.
+        // return the original option value attached with the event.
+        onChange({...event, value: optionValue});
```

Comment:
I seem to recall an issue with using `...event` like this in the past. Unfortunately, I cannot remember the details. But, I think we should keep the pattern of:
```
const adjustedEvent = event;
adjustedEvent.value = optionValue;
```

Reply:
Fixed!